### PR TITLE
Add Sleep testing and class

### PR DESCRIPTION
**Pull request by ohClaire**
# PR template

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

## Type of change

- [ ]  Bug fix (fixes an issue)
- [x]  New feature (adds functionality)
- [ ]  Refactor

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [ ]  I tested my changes in the browser
- [x]  I tested with Mocha/Chai

# Checklist:

- [x]  My code follows the style guidelines of this project
- [x]  I have performed a self-review of my own code
- [ ]  I have commented my code, particularly in hard-to-understand areas
- [ ]  I have made corresponding changes to the documentation
- [ ]  My changes generate no new bugs
- [x]  I have added tests that prove my fix is effective or that my feature works
- [x]  New and existing unit tests pass locally with my changes
- [x]  Is your code D.R.Y?
- [x]  Does it follow SRP?

### CODE REVIEW

File: src/Sleep.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    sum = dataSet.reduce((acc, data) => {
+      if (data[type]) {
+        acc += data[type];
+      }
+
+      return acc;
+    }, 0);
+
+    average = sum / dataSet.length;
+    return Number(average.toFixed(1));
```
Comment (emilyjmiles) [inline src/Sleep.js]: Can you change the return to Math.round(sum / dataSet.length) to dry this function up a little?

**Review by emilyjmiles — CHANGES_REQUESTED** [review]
Missing some semicolons.

**Review by emilyjmiles — APPROVED** [review]
Thank you for resolving the changes request!

**Review by coleanthony1990 — APPROVED** [review]
looks awesome!

**Pull request merged**